src/fluree/db/ledger/transact/txfunction.clj

```diff
@@ -0,0 +1,18 @@
+(ns fluree.db.ledger.transact.txfunction
+  (:require [fluree.db.dbfunctions.core :as dbfunctions]))
+
+;; functions related to transaction functions
+
+;; TODO - can probably parse function string to final 'lisp form' when generating TxFunction
+(defrecord TxFunction [fn-str])
```

Comment:
It allows for polymorphism without recasting all values as a map so we can embed a type.  I suppose it could be accomplished in different ways - but this seemed efficient (we don't recast all values, just tx-functions), and keeps logic clear downstream as we can simply do (tx-function? val). I was thinking we'd have other values we'd cast as types too, just started here.